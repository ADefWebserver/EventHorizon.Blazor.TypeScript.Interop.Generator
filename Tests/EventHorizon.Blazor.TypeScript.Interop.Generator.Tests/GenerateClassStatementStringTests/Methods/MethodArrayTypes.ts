declare module Static.Primitive {
    export class ExampleClass {
        arrayExample(): Array<string>;
        arrayLikeExample(): ArrayLike<string>;
        floatArrayExample(): FloatArray;
        float32ArrayExample(): Float32Array;
        indicesArrayExample(): IndicesArray;
        uint8ArrayExample(): Uint8Array;

        arrayExample(): Array<string>;
        arrayLikeExample(): ArrayLike<string>;
        floatArrayExample(): Nullable<FloatArray>;
        float32ArrayExample(): Nullable<Float32Array>;
        indicesArrayExample(): Nullable<IndicesArray>;
        uint8ArrayExample(): Nullable<Uint8Array>;
        deepImmutableArray(): DeepImmutable<Float32Array>;
    }
}
