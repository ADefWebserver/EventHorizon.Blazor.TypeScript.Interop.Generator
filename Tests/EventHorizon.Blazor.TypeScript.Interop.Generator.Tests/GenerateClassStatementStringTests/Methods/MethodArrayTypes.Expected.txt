/// Generated - Do Not Edit
namespace Static.Primitive
{
    using System;
    using System.Collections.Generic;
    using System.Text.Json.Serialization;
    using System.Threading.Tasks;
    using EventHorizon.Blazor.Interop;
    using Microsoft.JSInterop;

    
    
    [JsonConverter(typeof(CachedEntityConverter))]
    public class ExampleClass : CachedEntityObject
    {
        #region Static Accessors

        #endregion

        #region Static Properties

        #endregion

        #region Static Methods

        #endregion

        #region Accessors

        #endregion

        #region Properties

        #endregion
        
        #region Constructor
        public ExampleClass() : base() { } 

        public ExampleClass(
            ICachedEntity entity
        ) : base(entity)
        {
            ___guid = entity.___guid;
        }


        #endregion

        #region Methods
        public string[] arrayExample()
        {
            return EventHorizonBlazorInteropt.FuncArray<string>(
                new object[]
                {
                    new string[] { this.___guid, "arrayExample" }
                }
            );
        }

        public string[] arrayLikeExample()
        {
            return EventHorizonBlazorInteropt.FuncArray<string>(
                new object[]
                {
                    new string[] { this.___guid, "arrayLikeExample" }
                }
            );
        }

        public decimal[] floatArrayExample()
        {
            return EventHorizonBlazorInteropt.FuncArray<decimal>(
                new object[]
                {
                    new string[] { this.___guid, "floatArrayExample" }
                }
            );
        }

        public decimal[] float32ArrayExample()
        {
            return EventHorizonBlazorInteropt.FuncArray<decimal>(
                new object[]
                {
                    new string[] { this.___guid, "float32ArrayExample" }
                }
            );
        }

        public decimal[] indicesArrayExample()
        {
            return EventHorizonBlazorInteropt.FuncArray<decimal>(
                new object[]
                {
                    new string[] { this.___guid, "indicesArrayExample" }
                }
            );
        }

        public decimal[] uint8ArrayExample()
        {
            return EventHorizonBlazorInteropt.FuncArray<decimal>(
                new object[]
                {
                    new string[] { this.___guid, "uint8ArrayExample" }
                }
            );
        }

        public decimal[] deepImmutableArray()
        {
            return EventHorizonBlazorInteropt.FuncArray<decimal>(
                new object[]
                {
                    new string[] { this.___guid, "deepImmutableArray" }
                }
            );
        }
        #endregion
    }
}